src/pages/nfts/collections/[collectionAddress]/[tokenId].tsx

```diff
@@ -1,11 +1,65 @@
 import IndividualNFT from 'views/Nft/market/Collection/IndividualNFTPage'
-import { NftMarketLayout } from 'views/Nft/market/Layout'
 import React from 'react'
+import { GetStaticPaths, GetStaticProps, InferGetStaticPropsType } from 'next'
+import { getNftApi } from 'state/nftMarket/helpers'
+import { NftToken } from 'state/nftMarket/types'
+// eslint-disable-next-line camelcase
+import { SWRConfig, unstable_serialize } from 'swr'
 
-const IndividualNFTPage = () => {
-  return <IndividualNFT />
+const IndividualNFTPage = ({ fallback }: InferGetStaticPropsType<typeof getStaticProps>) => {
+  return (
+    <SWRConfig
+      value={{
+        fallback,
+      }}
+    >
+      <IndividualNFT />
+    </SWRConfig>
+  )
 }
 
-IndividualNFTPage.Layout = NftMarketLayout
+export const getStaticPaths: GetStaticPaths = async () => {
+  return {
+    fallback: true,
+    paths: [],
+  }
+}
+
+export const getStaticProps: GetStaticProps = async ({ params }) => {
+  const { collectionAddress, tokenId } = params
+
+  if (typeof collectionAddress !== 'string' || typeof tokenId !== 'string') {
+    return {
+      notFound: true,
+    }
+  }
+
+  const metadata = await getNftApi(collectionAddress, tokenId)
+  if (!metadata) {
+    return {
+      notFound: true,
+      revalidate: 1,
+    }
+  }
+
+  const nft: NftToken = {
+    tokenId,
+    collectionAddress,
+    collectionName: metadata.collection.name,
+    name: metadata.name,
+    description: metadata.description,
+    image: metadata.image,
+    attributes: metadata.attributes,
+  }
+
+  return {
+    props: {
+      fallback: {
+        [unstable_serialize(['nft', nft.collectionAddress, nft.tokenId])]: nft,
+      },
+    },
+    revalidate: 60 * 60 * 12, // 12 hours
```

Comment:
Some of the nft collections shows different images when pre-minted nfts is not sold yet to the users. (Such as catbread afaik). That means user has to wait 12 hours to see the final image?